rust/metaplex/program/src/utils.rs

```diff
@@ -604,7 +606,19 @@ pub fn common_redeem_checks(
     }
 
     if AuctionData::get_state(auction_info)? != AuctionState::Ended {
-        return Err(MetaplexError::AuctionHasNotEnded.into());
+        if win_index.is_some() && auction_extended_info.is_some() {
```

Comment:
To keep it clean let's abstract this into it's own method. Something like assert_auction_is_ended_or_this_is_valid_instant_sale()?

Reply:
In the case of open edition auctions, win_index may not be some. It will be none. That should still work just fine. Have you considered the OpenEdition auction type? Because that's the case of selling an infinite number of items at 5SOL each :)